C++/math/Ackermann_function.cpp

```diff
@@ -0,0 +1,28 @@
+// Ackermann Function implemented in C++
+
+#include <iostream> 
+using namespace std; 
+  
+int ackermann_function(int m, int n) 
+{ 
+    if (m == 0){ 
+        return n + 1; 
+    } 
+    else if((m > 0) && (n == 0)){ 
+        return ackermann_function(m - 1, 1); 
+    } 
+    else if((m > 0) && (n > 0)){ 
+        return ackermann_function(m - 1, ackermann_function(m, n - 1)); 
+    } 
+} 
+  
+// Driver code 
+int main() 
+{ 
+    int m,n;
+    cout<<"Enter two non negative integers:"<<endl;
+    cin>>m>>n;
+    int result=ackermann_function(m,n);
```

Comment:
@YashThakare add time and space complexity in the code comments and update the readme too.